esp-hal/src/interrupt/mod.rs

```diff
@@ -225,10 +225,6 @@ impl Iterator for InterruptStatusIterator {
     type Item = u8;
 
     fn next(&mut self) -> Option<Self::Item> {
-        if self.idx == usize::MAX {
```

Comment:
I don't recall why this was here in the first place, but why is it safe to remove?

Reply:
First up, we don't poll the iterator after it returns None, so this `if` is never practically hit.

Second, the for loop would just be skipped if `idx >= STATUS_WORDS`.